SU2_CFD/src/output/filewriter/CSTLFileWriter.cpp

```diff
@@ -0,0 +1,348 @@
+  if (sorted_halo_nodes.empty()) sorted_halo_nodes.resize(1); /* Avoid crash. */
+  SU2_MPI::Alltoallv(&sorted_halo_nodes[0], &num_nodes_to_receive[0], &nodes_to_receive_displacements[0], MPI_UNSIGNED_LONG,
+                     &nodes_to_send[0],     &num_nodes_to_send[0],    &nodes_to_send_displacements[0],    MPI_UNSIGNED_LONG,
+                     MPI_COMM_WORLD);
+
+  /* Now actually send and receive the data */
+  data_to_send.resize(max(1, total_num_nodes_to_send * (int)fieldnames.size()));
+  halo_var_data.resize(max(static_cast<unsigned long>(1), fieldnames.size() * num_halo_nodes));
+  num_values_to_send.resize(size);
+  values_to_send_displacements.resize(size);
+  num_values_to_receive.resize(size);
+  unsigned long index = 0;
+
+  for(int iRank = 0; iRank < size; ++iRank) {
+
+    /* We send and receive GlobalField_Counter values per node. */
+    num_values_to_send[iRank]              = num_nodes_to_send[iRank] * fieldnames.size();
+    values_to_send_displacements[iRank]    = nodes_to_send_displacements[iRank] * fieldnames.size();
+    num_values_to_receive[iRank]           = num_nodes_to_receive[iRank] * fieldnames.size();
+    values_to_receive_displacements[iRank] = nodes_to_receive_displacements[iRank] * fieldnames.size();
+
+    for(iVar = 0; iVar < fieldnames.size(); ++iVar) {
+      for(int iNode = 0; iNode < num_nodes_to_send[iRank]; ++iNode) {
+        unsigned long node_offset = nodes_to_send[nodes_to_send_displacements[iRank] + iNode] - dataSorter->GetNodeBegin(rank);
+        data_to_send[index++] = SU2_TYPE::GetValue(dataSorter->GetData(iVar,node_offset));
+      }
+    }
+
+  }
+
+  SU2_MPI::Alltoallv(&data_to_send[0],  &num_values_to_send[0],    &values_to_send_displacements[0],    MPI_DOUBLE,
+                     &halo_var_data[0], &num_values_to_receive[0], &values_to_receive_displacements[0], MPI_DOUBLE,
+                     MPI_COMM_WORLD);
+}
+
+
+void CSTLFileWriter::GatherCoordData(){
+  /*--- 2. Load the coordinate data succesively in an array. 3coordinates*3nodes = 9 consecutive
+    su2doubles form a triangle ---*/
+
+  /*--- Routine to write the surface STL files (ASCII). We
+   assume here that, as an ASCII file, it is safer to merge the
+   surface data onto the master rank for writing for 2 reasons:
+   (a) as a surface file, the amount of data should be much less
+   than the volume solution, and (b) writing ASCII files in parallel
+   requires serializing the IO calls with barriers, which ruins
+   the performance at moderate to high rank counts. ---*/
+
+  unsigned short iPoint,
+                 iVar;
+
+  unsigned long iProcessor,
+                nProcessor = size,
+                iElem,
+                nLocalTria,
+                nLocalQuad,
+                nLocalTriaAll = 0,
+                global_node_number;
+
+  su2double *bufD_Send = NULL;
+
+  vector<unsigned short> Nodelist = {0,1,3, 1,2,3}; // for Quad2Tri, assumes clockwise or counterclockwise rotation

```

Comment:
This is static sized, you do not need to use `vector`

Reply:
this code bit also does not exist anymore